server/auth_server.go

```diff
@@ -370,6 +370,22 @@ func updateSettings(saveConfig map[string]map[string]string) error {
 		// authConfig now was prevConfig
 		// saveConfig is to be saved, so authConfig should get added values from saveConfig
 		log.Debugf("Previous saved auth config: %v", authConfig)
+		// saveConfig (updated config) might contain no secret settings, if GetConfig(listOnly=true) was called before it
```

Comment:
Why would GetConfig(listOnly=true) save any config to the in-memory provider? GetConfig() is a read API, should not be saving back anything. I think it is the UpdateConfig that initializes a provider with a config having no secrets.